### PR TITLE
WT-13150 Refactor some useful functionality from the Python tools into tools/py_common

### PR DESCRIPTION
**Pull request by pmacko86**
This PR mostly ended up refactoring wt_binary_decode.py, moving some of its functionality into tools/py_common, especially integer decoding, the binary file wrapper, and much of the page, block, and cell decoding. I'm not sure if the latter was necessary, but I'd like to believe that this made wt_binary_decode a bit cleaner.

I also renamed a few things to give it more descriptive names and/or bring the field names closer to their original WT names.

### CODE REVIEW

**Comment by pmacko86** [comment]
Don, if you have better suggestions for names of the new py_common packages that I introduced, please suggest :) I feel like they could be better. Maybe `binary_format` could be at least renamed to `file_format` or `wt_format`?

Today I also came up with more things that I would like to refactor, but I'm thinking it would be better to wait to do it in another ticket so that I'd do this refactoring over more PRs rather than one giant PR. And also to make sure that you agree with the new direction before I do any more of this.

Thanks!

File: tools/py_common/binary_data.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+    '''
+
+    def __init__(self, fileobj):
+        '''
+        Initialize the instance of this binary stream.
+        '''
+        self.fileobj = fileobj
+        self.saved = bytearray()
+
+
```
Comment (ddanderson) [inline tools/py_common/binary_data.py:180]: Isn't a single blank line between functions/methods recommended?  (Maybe double blank between classes?). This file uses both 1 line and 2 lines between functions/methods.

File: tools/py_common/binary_data.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    def read_packed_uint64_with_size(self) -> typing.Tuple[int, int]:
+        '''
+        Read the next packed integer; return (integer, number of bytes read).
+        '''
+        start = self.tell()
+        arr = FileAsArray(self)
+        val, arr = unpack_int(arr)
+        sz = self.tell() - start
+        return (val, sz)
+
```
Comment (ddanderson) [inline tools/py_common/binary_data.py:238]: Nice encapsulation with these additional functions.

**Comment by ddanderson** [comment]
> Don, if you have better suggestions for names of the new py_common packages that I introduced, please suggest :) I feel like they could be better. Maybe `binary_format` could be at least renamed to `file_format` or `wt_format`?

Perhaps `btree_disk_format`?  Or `btree_format`?

**Review by ddanderson — APPROVED** [review]
Beautiful work, I love the new separation between parsing and printing.

File: tools/py_common/binary_data.py (hunk 1 above)
Comment (pmacko86) [reply tools/py_common/binary_data.py:180]: Good point—I'll adjust this to what the style guide says:

> Surround top-level function and class definitions with two blank lines.
> 
> Method definitions inside a class are surrounded by a single blank line.

Source: https://peps.python.org/pep-0008/

**Pull request merged**